### PR TITLE
Add regular expression for duration strings

### PR DESCRIPTION
**Pull request by Planeshifter**
Resolves # .

## Description

> What is the purpose of this pull request?

This pull request:

-   adds a regular expression for duration strings

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Planeshifter** [comment]
Pushed initial implementation. One open question, @kgryte: Currently, the regular expression positively validates an empty string. That should be fine with the existing packages for duration strings, but may or may not be desired. I think it's fine as-is, any objections?

**Comment by kgryte** [comment]
@Planeshifter Matching an empty string should be fine. No objections from me.

**Comment by Planeshifter** [comment]
Great! Package is ready for review now.

File: lib/node_modules/@stdlib/regexp/duration-string/docs/repl.txt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    A duration string is a string containing a sequence of time units. A time
+    unit is a nonnegative integer followed by a unit identifier. The following
+    unit identifiers are supported:
+
+    -   d: days
+    -   h: hours
+    -   m: minutes
+    -   s: seconds
+    -   ms: milliseconds
```
Comment (kgryte) [inline lib/node_modules/@stdlib/regexp/duration-string/docs/repl.txt]:
```suggestion
    - d: days
    - h: hours
    - m: minutes
    - s: seconds
    - ms: milliseconds
```

File: lib/node_modules/@stdlib/regexp/duration-string/docs/repl.txt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    Duration strings are case insensitive. For example, the string `1M3S10MS` is
+    equivalent to `1m3s10ms`.
+
+    The regular expression captures the following groups:
+
+    1.  The days component.
+    2.  The hours component.
+    3.  The minutes component.
+    4.  The seconds component.
+    5.  The milliseconds component.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/regexp/duration-string/docs/repl.txt]:
```suggestion
    1. The days component.
    2. The hours component.
    3. The minutes component.
    4. The seconds component.
    5. The milliseconds component.
```

**Review by kgryte — APPROVED** [review]

**Pull request merged**